### PR TITLE
add docker

### PR DESCRIPTION
**Pull request by Leanchers**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url"  : "jdbc:mysql://mysqldb:${MYSQLDB_DOCKER_PORT}/${MYSQLDB_DATABASE}",
+        "spring.datasource.username" : "${MYSQLDB_USER}",
+        "spring.datasource.password" : "${MYSQLDB_ROOT_PASSWORD}",
+        "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQLDialect",
+        "spring.datasource.driver-class-name" : "com.mysql.cj.jdbc.Driver"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (okuzan) [inline docker-compose.yml]:
```suggestion
      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:${DEBUG_PORT}"

```

File: .env [hunk 2]
```diff
@@ -0,0 +1,9 @@
+MYSQLDB_USER=root
+MYSQLDB_ROOT_PASSWORD=1234
+MYSQLDB_DATABASE=book_store
+MYSQLDB_LOCAL_PORT=3307
+MYSQLDB_DOCKER_PORT=3306
+
+SPRING_LOCAL_PORT=8088
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (okuzan) [inline .env]: let's add .env to .gitignore
rename this file to .env.sample
actual .env should never be in source control, it's supposed to contain secrets

**Review by okuzan — CHANGES_REQUESTED** [review]

File: .env.sample [hunk 3]
```diff
@@ -0,0 +1,9 @@
+MYSQLDB_USER=root
```
Comment (Elena-Bruyako) [inline .env.sample]:
```suggestion
MYSQLDB_USER=
```

remove other properties, it should be like template

File: .env.sample [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+MYSQLDB_ROOT_PASSWORD=1234
+MYSQLDB_DATABASE=book_store
+MYSQLDB_LOCAL_PORT=3307
+MYSQLDB_DOCKER_PORT=3306
+
+SPRING_LOCAL_PORT=8088
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (Elena-Bruyako) [inline .env.sample]: Remove .env file

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**